### ISSUE TITLE
Validate nonlinear Python closed-loop agent policy

### ISSUE THREAD

**Issue by Ahmadfauzi34** [Ahmadfauzi34/ai_agent_builder#231]
## Purpose

Follow #229/#230 after Python ABI v1 gains minimal ReLU layer-spec construction. Validate a genuinely nonlinear policy in the existing deterministic closed-loop control workload before adding any more ABI/model surface.

## Experimental control

Keep the established `research_python_agent_control_rollout.py` workload semantics unchanged:

- same deterministic 2D dynamics and disturbance functions;
- same four scenarios;
- same 16-step horizon;
- same reward/cost terms;
- same seed, OpenES strategy, population, sigma, learning rate, and generation count;
- Python continues to own environment, reward, evaluation schedule, and horizon;
- graph owns policy execution only;
- ES owns candidate generation/update only.

Change only policy representability:

```text
existing control:
Linear(6 -> 2, bias=true)

nonlinear research:
Linear(6 -> 8, bias=true)
  -> ReLU
  -> Linear(8 -> 2, bias=true)
```

Canonical parameter dimension must be exactly 74.

## Required proof

1. fresh installed wheel outside the repo, no `PYTHONPATH`;
2. typed host surface only for normal workload execution;
3. `GraphParameterBinding.total_len == 74` and canonical identities remain stable;
4. `ask_f32()` yields contiguous native-f32 candidate batches and candidate windows feed `apply_flat` directly;
5. all fitness/reward values remain finite across multiple generations;
6. stateful ProgramBundle export/import replays exact trainable state and final reward;
7. report ask/apply/rollout/tell timing without CI performance thresholds;
8. report zero-policy, initial champion, final champion, generation-best/champion histories;
9. compare descriptively with the established Linear-policy evidence, but do not encode a brittle superiority threshold or alter environment semantics to favor the nonlinear policy.

## Interpretation

- nonlinear policy trains cleanly and improves the same workload -> nonlinear consumer path is operational; use timing/evidence to choose the next system-level bottleneck;
- lifecycle/replay/identity breaks -> fix semantics before broadening model surface;
- training is materially harder despite sound semantics -> inspect optimizer/search scaling before adding more activation variants;
- no demonstrated benefit on this simple control task -> keep ReLU support for representability but do not widen activation surface solely for capability claims.

## Non-goals

No new ABI symbols, no additional activation variants, no graph/controller semantics, no optimizer algorithm changes, no Linear wrapper optimization, no zero-copy/NumPy/DLPack/PyO3, no Math Program v10, and no Go/C++ work.

Implementation should be research-only after #230 is merged.
